### PR TITLE
Doc for latest changes in blitz

### PR DESCRIPTION
**Pull request by KamyarZiabari**

### CODE REVIEW

File: develop-guide/writeblitz/writeblitz.rst [hunk 1]
```diff
@@ -164,6 +177,25 @@ exists in the output. You also, have the option to check if a specific
 
 Both include and exclude keywords are optional to use.
 
+.. note::
+
+    The all the arguments that can be sent to the device while executing a command on a device (specified in this `link
```
Comment (jeaubin) [inline develop-guide/writeblitz/writeblitz.rst:182]: All the arguments for the execute command can be found at this link.

File: develop-guide/writeblitz/writeblitz.rst [hunk 2]
```diff
@@ -191,19 +223,6 @@ this `section
 
         ...
 
-configure
```
Comment (jeaubin) [inline develop-guide/writeblitz/writeblitz.rst:194]: why move configure? having it after execute make sense

File: develop-guide/writeblitz/writeblitz.rst [hunk 3]
```diff
@@ -398,17 +417,17 @@ and later on use that values in other actions
...
                 - variable_name: everything
               commands:
                 - pwd
                 - ls
                 - |
-                  cd ~  
+                  cd ~
                   echo A string of text
+              include: 
+                  - contains('ls')
```
Comment (jeaubin) [inline develop-guide/writeblitz/writeblitz.rst:430]: ls is a linux command,  

what about cat file.txt

and contains('some text which is contained in file.txt')

File: develop-guide/writeblitz/writeblitz.rst [hunk 4]
```diff
@@ -550,6 +569,41 @@ Action ``compare`` allows you to verify the values of the saved variables. Below
         - "'%VARIABLES{os}' == 'NX-OS' and '%VARIABLES{date_created}' == '10/22/2019 10:00:00 [10/22/2019 16:57:31]'"
         - " %VARIABLES{bootflash} >= 290000 or '%VARIABLES{bios}' == '07.33'"
 
+
+
+.. note::
+
+    The starting message of an step in etrade logs, is predefined. However, that value can be changed.
```
Comment (jeaubin) [inline develop-guide/writeblitz/writeblitz.rst:576]: trade is not released externally, cant say this.

The starting message of a Step can be modified by using ....

File: develop-guide/writeblitz/writeblitz.rst [hunk 5]
```diff
@@ -550,6 +569,41 @@ Action ``compare`` allows you to verify the values of the saved variables. Below
...
+    In your action you can specify your custom message like the example below and change the starting action log message.
+
+.. code-block:: YAML
+
+    - execute:
+        command: show version
+        device: PE1
+        custom_step_message: This is a test to see if a custom_step_message would be applied
```
Comment (jeaubin) [inline develop-guide/writeblitz/writeblitz.rst:584]: is this the start or end.

change the variable name to custom_start_step_message 

what do you think

File: develop-guide/writeblitz/writeblitz.rst [hunk 6]
```diff
@@ -550,6 +569,41 @@ Action ``compare`` allows you to verify the values of the saved variables. Below
         - "'%VARIABLES{os}' == 'NX-OS' and '%VARIABLES{date_created}' == '10/22/2019 10:00:00 [10/22/2019 16:57:31]'"
         - " %VARIABLES{bootflash} >= 290000 or '%VARIABLES{bios}' == '07.33'"
 
+
+
+.. note::
```
Comment (jeaubin) [inline develop-guide/writeblitz/writeblitz.rst:574]: why is this note in compare?

File: develop-guide/writeblitz/writeblitz.rst [hunk 7]
```diff
@@ -550,6 +569,41 @@ Action ``compare`` allows you to verify the values of the saved variables. Below
...
+        timeout: 100
+
+as shown in the image you can see how in the logs the starting message is customized.
+
+.. image:: ../images/custom_step_msg.png
+   :width: 200%
+
+.. note::
+
+    You can get a Passed result for an action that is expected to fail by using setting the key; ``expected_failure: True``.
```
Comment (jeaubin) [inline develop-guide/writeblitz/writeblitz.rst:594]: why is this here

you sould have a section named; negative testing adn explain it there   not sure a small note in configure replace

File: develop-guide/writeblitz/writeblitz.rst [hunk 8]
```diff
@@ -550,6 +569,41 @@ Action ``compare`` allows you to verify the values of the saved variables. Below
...
+
+.. code-block:: YAML
+
+    # The command doesnt exist so action should error out but since it was anticipated that the command wouldn't work.
```
Comment (jeaubin) [inline develop-guide/writeblitz/writeblitz.rst:598]: does this work for error too? Not just for failure?

**Review by jeaubin — CHANGES_REQUESTED** [review]

File: develop-guide/writeblitz/writeblitz.rst (hunk 1 above)
Comment (KamyarZiabari) [reply develop-guide/writeblitz/writeblitz.rst:182]: Done

File: develop-guide/writeblitz/writeblitz.rst (hunk 2 above)
Comment (KamyarZiabari) [reply develop-guide/writeblitz/writeblitz.rst:194]: Added after execute

File: develop-guide/writeblitz/writeblitz.rst (hunk 3 above)
Comment (KamyarZiabari) [reply develop-guide/writeblitz/writeblitz.rst:430]: to search within that file is not supported, as its looks into this just like a dict. They can do contains('text in the file', regex=True), but that is on its own is limited. I will address a better solution for future

File: develop-guide/writeblitz/writeblitz.rst (hunk 4 above)
Comment (KamyarZiabari) [reply develop-guide/writeblitz/writeblitz.rst:576]: addressed

File: develop-guide/writeblitz/writeblitz.rst (hunk 5 above)
Comment (KamyarZiabari) [reply develop-guide/writeblitz/writeblitz.rst:584]: addressed

File: develop-guide/writeblitz/writeblitz.rst (hunk 6 above)
Comment (KamyarZiabari) [reply develop-guide/writeblitz/writeblitz.rst:574]: Its not in compare its at the end the list of actions

File: develop-guide/writeblitz/writeblitz.rst (hunk 7 above)
Comment (KamyarZiabari) [reply develop-guide/writeblitz/writeblitz.rst:594]: Done

File: develop-guide/writeblitz/writeblitz.rst (hunk 8 above)
Comment (KamyarZiabari) [reply develop-guide/writeblitz/writeblitz.rst:598]: Errors that we are handling yes. but if the user is missing to put the device in the action. No its not supported for that. Should we support that ?! no right?

File: develop-guide/writeblitz/writeblitz.rst (hunk 3 above)
Comment (jeaubin) [reply develop-guide/writeblitz/writeblitz.rst:433]: Then what does include search for? Not what it prints the screen  just the commadn you send?

File: develop-guide/writeblitz/writeblitz.rst (hunk 5 above)
Comment (jeaubin) [reply develop-guide/writeblitz/writeblitz.rst]: in the code too right ?

File: develop-guide/writeblitz/writeblitz.rst (hunk 8 above)
Comment (jeaubin) [reply develop-guide/writeblitz/writeblitz.rst:602]: In my mind, it should only for failure  not error

File: develop-guide/writeblitz/writeblitz.rst [hunk 9]
```diff
@@ -177,16 +164,32 @@ exists in the output. You also, have the option to check if a specific
...
+            shutdown
+
 .. note::
 
-    The all the arguments that can be sent to the device while executing a command on a device (specified in this `link
-    <http://wwwin-pyats.cisco.com/cisco-shared/unicon/latest/user_guide/services/generic_services.html#execute>`_.)
-    and while configuring a device (specified in this `link
-    <http://wwwin-pyats.cisco.com/cisco-shared/unicon/latest/user_guide/services/generic_services.html#configure>`_.) 
-    can be used like below example.
+    You can apply additional arguments to commands ``execute`` and ``configure``.
```
Comment (jeaubin) [inline develop-guide/writeblitz/writeblitz.rst:182]: why isnt execute with the execute.

it makes no sense to show execute in the configure section

File: develop-guide/writeblitz/writeblitz.rst (hunk 5 above)
Comment (KamyarZiabari) [reply develop-guide/writeblitz/writeblitz.rst]: Yes

File: develop-guide/writeblitz/writeblitz.rst (hunk 8 above)
Comment (KamyarZiabari) [reply develop-guide/writeblitz/writeblitz.rst:602]: There is only one error, that we are handeling that I set this to pass, which is for api error, on function name or args. I thought in case they have doubts about the function name maybe but they want to give it try. Just above that there is an error, that I didnt apply this to
image

File: develop-guide/writeblitz/writeblitz.rst (hunk 6 above)
Comment (jeaubin) [reply develop-guide/writeblitz/writeblitz.rst:576]: What about creating a section called Tips and put all of them there

File: develop-guide/writeblitz/writeblitz.rst [hunk 10]
```diff
@@ -570,28 +573,29 @@ Action ``compare`` allows you to verify the values of the saved variables. Below
...
+    The starting message of a Step can be modified by specifying a custom message like the example below. This can be applied
+    to all the actions supported in Blitz.
 
 .. code-block:: YAML
 
     - execute:
         command: show version
         device: PE1
-        custom_step_message: This is a test to see if a custom_step_message would be applied
+        custom_start_step_message: This is a test to see if a custom_start_step_message would be applied
```
Comment (jeaubin) [inline develop-guide/writeblitz/writeblitz.rst:586]: Not a good message for a doc; if you are writing doc it better work :D 

        custom_start_step_message: My own message instead of the default one

File: develop-guide/writeblitz/writeblitz.rst (hunk 6 above)
Comment (KamyarZiabari) [reply develop-guide/writeblitz/writeblitz.rst:576]: like all the notes?

File: develop-guide/writeblitz/writeblitz.rst (hunk 10 above)
Comment (KamyarZiabari) [reply develop-guide/writeblitz/writeblitz.rst:586]: Done

File: develop-guide/writeblitz/writeblitz.rst (hunk 9 above)
Comment (KamyarZiabari) [reply develop-guide/writeblitz/writeblitz.rst:182]: They were back to back actions, and technically same feature, thought would be more efficient to have them together. Changing to your way

Comment (KamyarZiabari) [reply develop-guide/writeblitz/writeblitz.rst:182]: Done

File: develop-guide/writeblitz/writeblitz.rst (hunk 3 above)
Comment (KamyarZiabari) [reply develop-guide/writeblitz/writeblitz.rst:433]: A dict contains of a pair of command and output to that command. all of them are stored to a dict, include/exclude would treat it like a dict.

**Review by jeaubin — APPROVED** [review]
